### PR TITLE
Feature/error 404 dynamic router

### PR DESCRIPTION
**Pull request by mv-v-7**
When you made review please also think about route system

### CODE REVIEW

File: client/src/components/guard/RoutersTesterGuard.vue [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+export default {
+    name: 'RoutersTesterGuard',
+
+    async beforeRouteEnter(to, from, next) {
+        try {
+            await routersTesterService.checkNickname(to.params.nickname);
+        } catch (e) {
+            next({ name: 'Error404' });
+        }
+        next();
```
Comment (dmitriy-uvin) [inline client/src/components/guard/RoutersTesterGuard.vue:17]: move `next()` into `try`

File: client/src/router/index.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ import qs from 'qs';
 import UserDataProvider from '@/components/guard/UserDataProvider';
 import LoginGuard from '@/components/guard/LoginGuard';
 import AuthGuard from '@/components/guard/AuthGuard';
+import RoutersTesterGuard from '../components/guard/RoutersTesterGuard';
```
Comment (dmitriy-uvin) [inline client/src/router/index.js:8]: rename `RoutersTesterGuard` to `UserExistGuard` or smth like that

Comment (mv-v-7) [reply client/src/router/index.js:8]: In the future, I should check if :nickname/:id  For this I should create another Guard?

File: client/src/store/modules/i18n/en.js [hunk 3]
```diff
@@ -397,5 +397,12 @@ export default {
...
     PUBLIC_LINK_WAS_COPIED: 'Public Link was copied!',
     TAGS: 'Tags',
-    ALL_TAGS: 'All Tags'
+    ALL_TAGS: 'All Tags',
+    ERROR_404: 'Error 404',
+    THE_PAGE_YOU_ARE_LOOKING_FOR_COULD_NOT_BE_FOUND:
+        'The page you are looking for could not be found',
+    PLEASE_MAKE_SURE_THAT_YOU_TYPED_IN_THE_URL_CORRECTLY:
+        " Please make sure that you've typed in the URL correctly",
+    PLEASE_RETYPE_ADDRESS_OR_RETURN_TO: 'Please retype address or return to',
```
[the remark this answers is not in the document]
Comment (mv-v-7) [reply client/src/store/modules/i18n/en.js:406]: Main page add in another constant for creating link

File: client/src/store/modules/i18n/uk.js [hunk 4]
```diff
@@ -316,5 +316,13 @@ export default {
...
     PUBLIC_LINK_WAS_COPIED: 'Публічне посилання було скопійоване!',
     TAGS: 'Теги',
-    ALL_TAGS: 'Усі теги'
+    ALL_TAGS: 'Усі теги',
+    ERROR_404: 'Помилка 404',
+    THE_PAGE_YOU_ARE_LOOKING_FOR_COULD_NOT_BE_FOUND:
+        'Сторінку, яку ви шукаєте, не вдалося знайти',
+    PLEASE_MAKE_SURE_THAT_YOU_TYPED_IN_THE_URL_CORRECTLY:
+        ' Переконайтесь, що ви правильно ввели URL-адресу',
+    PLEASE_RETYPE_ADDRESS_OR_RETURN_TO:
```
[the remark this answers is not in the document]
Comment (mv-v-7) [reply client/src/store/modules/i18n/uk.js:325]: Main page added as separate  constant for creating link

File: api/app/Http/Controllers/Api/RoutersTester.php [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Actions\RoutersTester\CheckNicknameAction;
+use App\Actions\RoutersTester\CheckNicknameRequest;
+
+class RoutersTester extends ApiController
```
[the remark this answers is not in the document]
Comment (mv-v-7) [reply api/app/Http/Controllers/Api/RoutersTester.php:8]: RoutersTester stand for checking all dynamic routers :nickname   :nickname/:id  :nickname/:id/:date and all future

File: client/src/router/index.js (hunk 2 above)
Comment (dmitriy-uvin) [reply client/src/router/index.js:8]: Maybe is, like EventTypeExistGuard

Comment (mv-v-7) [reply client/src/router/index.js:8]: made renaming maybe proposed name more meaning for developers

File: api/app/Http/Controllers/Api/RoutersTester.php [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+{
+    public function __construct()
```
[the remark this answers is not in the document]
Comment (mv-v-7) [reply api/app/Http/Controllers/Api/RoutersTester.php]: removed

File: client/src/components/guard/RoutersTesterGuard.vue (hunk 1 above)
Comment (mv-v-7) [reply client/src/components/guard/RoutersTesterGuard.vue]: fixed

**Review by dmitriy-uvin — APPROVED** [review]

**Pull request merged**